src/main/java/teammates/ui/webapi/action/DeleteFeedbackResponseCommentAction.java

```diff
@@ -36,6 +38,9 @@ public void checkSpecificAccessControl() {
 
         switch (intent) {
         case STUDENT_SUBMISSION:
+            if (userInfo == null && StringHelper.isEmpty(getRequestParamValue(Const.ParamsNames.REGKEY))) {
```

Comment:
The check can be (and should be) inside `getStudentOfCourseFromRequest` to save the duplicated code.